src/wp-includes/class-wp-meta-query.php

```diff
@@ -398,14 +398,16 @@ protected function get_sql_clauses() {
 	 * @param array $query Query to parse (passed by reference).
 	 * @param int   $depth Optional. Number of tree levels deep we currently are.
 	 *                     Used to calculate indentation. Default 0.
+	 * @param bool  $has_or_condition Optional. Whether the query contains any OR comparisons.
```

Comment:
Ah, I hadn't spotted that. Looks like it might work - if so, that's definitely a nicer approach than passing this extra parameter around. I'll look into it, thanks.